frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -1,71 +1,126 @@
+  const updateRubric = (newCriteria: Criteria[]) => {
+    setRubric((prevRubric) =>
+      prevRubric
+        ? {
+            ...prevRubric,
+            criteria: [...prevRubric.criteria, ...newCriteria],
+            pointsPossible:
+              prevRubric.pointsPossible +
+              newCriteria.reduce((sum, criterion) => sum + criterion.points, 0),
+          }
+        : {
+            id: parseInt(uuid(), 16),
+            title: "Imported Rubric",
+            criteria: newCriteria,
+            pointsPossible: newCriteria.reduce(
+              (sum, criterion) => sum + criterion.points,
+              0,
+            ),
+            key: `rubric-${uuid()}`,
+          },
+    );
+  };
 
-        // Validate each row to ensure it matches CSVRow type
-        const parsedData = results.data.filter((row): row is CSVRow => {
-          return (
-            Array.isArray(row) &&
-            typeof row[0] === "string" &&
-            row
-              .slice(1)
-              .every(
-                (cell) => typeof cell === "string" || typeof cell === "number",
-              )
-          );
-        });
-        console.log("Validated CSV data:", results.data);
-        onDataChange(parsedData); // Pass validated data to parent
-      },
-    });
-    closeImportCard();
+  const handleVersionSelection = (version: "versionOne" | "versionTwo") => {
+    const fileInput = document.createElement("input");
+    fileInput.type = "file";
+    fileInput.accept = ".csv";
+    fileInput.onchange = (event: Event) => {
+      const target = event.target as HTMLInputElement;
+      const file = target.files?.[0];
+      if (file) {
+        handleFileChange(file, version);
+      }
+    };
+    fileInput.click();
+    setShowVersionModal(false);
   };
 
   return (
-    <div className={"flex justify-center items-center gap-10"}>
-      <input
-        type="file"
-        accept=".csv"
-        data-testid={"file-upload"}
-        onChange={handleFileChange}
-        className="mt-4 mb-4 border border-gray-600 rounded-lg p-3 text-gray-300 hover:bg-gray-800 transition duration-300 cursor-pointer focus:outline-none"
-      />
-
-      {/* Cancel Button */}
+    <div className="flex flex-col items-center gap-4">
       <button
-        onClick={closeImportCard}
-        className=" mt-4 bg-red-600 text-white font-bold rounded-lg py-2 px-4 transition duration-300 hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-red-500"
+        onClick={() => setShowVersionModal(true)}
+        className="bg-blue-600 text-white font-bold rounded-lg py-2 px-4 transition duration-300 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-blue-500"
       >
-        Cancel
+        Import CSV
       </button>
+      <Dialog
+        isOpen={showVersionModal}
+        onClose={() => setShowVersionModal(false)}
+        title="Select Import Version"
+      >
+        <div className="flex flex-col gap-4">
+          <button
```

Comment:
see named constants comment

Reply:
Created Constant for VERSION_ONE and VERSION_TWO